File: ssm-sdk-client/src/test/java/ssm/client/SsmClientOptionalTest.java
package ssm.client;


import org.junit.jupiter.api.*;
import ssm.dsl.SsmAgentBase;
import ssm.dsl.SsmSessionStateBase;
import ssm.dsl.SsmBase;

import java.util.Optional;
import java.util.UUID;
import java.util.concurrent.CompletableFuture;

import static org.assertj.core.api.Assertions.assertThat;

public class SsmClientOptionalTest {

    private static SsmClient client;

    @BeforeAll
    public static void init() throws Exception {
        client = SsmClientTestBuilder.build();
    }

    @Test
    public void getAdminUser() throws Exception {
        CompletableFuture<Optional<SsmAgentBase>> agentRet = client.getAdmin(UUID.randomUUID().toString());
        Optional<SsmAgentBase> agentFormClient = agentRet.get();
        assertThat(agentFormClient).isEmpty();
    }

    @Test
    public void getAgentUser2() throws Exception {
        CompletableFuture<Optional<SsmAgentBase>> agentRet = client.getAgent(UUID.randomUUID().toString());
        Optional<SsmAgentBase> agentFormClient = agentRet.get();
        assertThat(agentFormClient).isEmpty();
    }

    @Test
    public void getSsm() throws Exception {
        CompletableFuture<Optional<SsmBase>> ssmReq = client.getSsm(UUID.randomUUID().toString());
        Optional<SsmBase> ssm = ssmReq.get();
        assertThat(ssm).isEmpty();
    }

    @Test
    public void getSession() throws Exception {
        CompletableFuture<Optional<SsmSessionStateBase>> ses = client.getSession(UUID.randomUUID().toString());
        Optional<SsmSessionStateBase> sesReq = ses.get();
        assertThat(sesReq).isEmpty();
    }

}

File: ssm-sdk/ssm-sdk-core/src/test/kotlin/ssm/sdk/client/SsmClientOptionalTest.kt
package ssm.sdk.client

import java.util.UUID
import kotlinx.coroutines.runBlocking
import org.assertj.core.api.Assertions
import org.junit.jupiter.api.Test
import ssm.chaincode.dsl.model.uri.ChaincodeUri

class SsmClientOptionalTest {

	private var client = SsmClientTestBuilder.build().buildQueryService()

	private val chaincodeUri = ChaincodeUri("chaincode:sandbox:ssm")

	@Test
	fun adminUser(): Unit = runBlocking {
		val agentRet = client.getAdmin(chaincodeUri, UUID.randomUUID().toString())
		Assertions.assertThat(agentRet).isNull()
	}

	@Test
	fun agentUser2(): Unit = runBlocking {
		val agentRet = client.getAgent(chaincodeUri,UUID.randomUUID().toString())
		Assertions.assertThat(agentRet).isNull()
	}

	@Test
	fun ssm(): Unit = runBlocking {
		val ssmReq = client.getSsm(chaincodeUri, UUID.randomUUID().toString())
		Assertions.assertThat(ssmReq).isNull()
	}

	@Test
	fun session(): Unit = runBlocking {
		val ses = client.getSession(chaincodeUri, UUID.randomUUID().toString())
		Assertions.assertThat(ses).isNull()
	}
}
